File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task58.py

def XO(s):
    if s is None:
        return True
    
    lower_str = s.lower()
    count_x = 0
    count_o = 0
    
    for c in lower_str:
        if c == 'x':
            count_x += 1
        elif c == 'o':
            count_o += 1
    
    return count_x == count_o

if __name__ == "__main__":
    print("Test 1: XO('ooxx') =", XO("ooxx"))
    print("Test 2: XO('xooxx') =", XO("xooxx"))
    print("Test 3: XO('ooxXm') =", XO("ooxXm"))
    print("Test 4: XO('zpzpzpp') =", XO("zpzpzpp"))
    print("Test 5: XO('zzoo') =", XO("zzoo"))

File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task58.java

public class Task58 {
    public static boolean XO(String str) {
        if (str == null) {
            return true;
        }
        
        String lowerStr = str.toLowerCase();
        int countX = 0;
        int countO = 0;
        
        for (int i = 0; i < lowerStr.length(); i++) {
            char c = lowerStr.charAt(i);
            if (c == 'x') {
                countX++;
            } else if (c == 'o') {
                countO++;
            }
        }
        
        return countX == countO;
    }
    
    public static void main(String[] args) {
        System.out.println("Test 1: XO(\\"ooxx\\") = " + XO("ooxx"));
        System.out.println("Test 2: XO(\\"xooxx\\") = " + XO("xooxx"));
        System.out.println("Test 3: XO(\\"ooxXm\\") = " + XO("ooxXm"));
        System.out.println("Test 4: XO(\\"zpzpzpp\\") = " + XO("zpzpzpp"));
        System.out.println("Test 5: XO(\\"zzoo\\") = " + XO("zzoo"));
    }
}
